cf_xarray/accessor.py

```diff
@@ -168,6 +177,31 @@ def _get_groupby_time_accessor(var: Union[DataArray, Dataset], key: str) -> List
         return []
 
 
+def _get_custom_criteria(
+    obj: Union[DataArray, Dataset], key: str, criteria=None
+) -> List[str]:
+
+    if isinstance(obj, DataArray):
+        obj = obj._to_temp_dataset()
+
+    if criteria is None:
+        if not OPTIONS["custom_criteria"]:
+            return []
+        criteria = OPTIONS["custom_criteria"]
+
+    if criteria is not None:
+        criteria = always_iterable(criteria, allowed=(tuple, dict, set))
+
+    results: Set = set()
+    for crit in criteria:
+        if key in crit:
+            for criterion, expected in crit[key].items():
+                for var in obj.variables:
+                    if obj[var].attrs.get(criterion, None) in expected:
```

Comment:
will have to deal with regexes here. Maybe always treat `expected` as a regex and pass to `re.match`?

Also this will need to deal with `"name"` in the criteria. So something like 
``` python
if criterion == 'name' and var == expected:
	results.update((var,))
```

Reply:
Ok there are a couple more of your suggestions here that I will look at locally.

Reply:
Can you provide a brief example of how you imagine this working that I can test with locally? I'm struggling with making these suggestions concrete when I don't have an example to run and I am not sure how to come up with it on my own without the vision you have.

Reply:
Using your test dataset, we want this to work

```python
    my_custom_criteria = {
        "ssh": {
            "standard_name": (
                "sea_surface_height_*",
            ),
            "name": ("sea_surface_elevation",),  # variable name
        }
	}

	cfxr.set_options(my_custom_criteria)

	ds = xr.Dataset()
	ds["elev"] = ("dim", np.arange(10), {"standard_name": "sea_surface_height_asd"})
	ds["sea_surface_elevation"] = ("dim", np.arange(10))

	xr.testing.assert_identical(ds.cf["ssh"], ds["elev"])
	xr.testing.assert_identical(ds.cf["ssh"], ds["sea_surface_elevation"])
```

Does this look like what you want?

Reply:
Yes, I think that is a good example.